### PR TITLE
Persist history completion selection if it is still valid on source update

### PR DESCRIPTION
**Pull request by cmcaine**

### CODE REVIEW

File: src/completions/History.ts [hunk 1]
```diff
@@ -76,22 +76,29 @@ export class HistoryCompletionSource extends Completions.CompletionSourceFuse {
...
+        this.deselect()
 
-    updateChain() {
-        // Options are pre-trimmed to the right length.
-        this.options.forEach(option => (option.state = "normal"))
+        // Set initial state to normal, unless the option was selected a moment
+        // ago, then reselect it so that users don't lose their selections.
+        this.options.forEach(option => {
+            if (lastFocused !== undefined && lastFocused.value === option.value) {
+                this.select(option)
```
Comment (bovine3dom) [inline src/completions/History.ts]: Can we stick a `break` here to ensure we can only ever have one selected?

File: src/completions/History.ts [hunk 2]
```diff
@@ -76,22 +76,29 @@ export class HistoryCompletionSource extends Completions.CompletionSourceFuse {
...
+            } else
+                option.state = "normal"
```
Comment (bovine3dom) [inline src/completions/History.ts]: Obviously we'd need to do a first pass with this on everything

File: src/completions/History.ts [hunk 3]
```diff
@@ -87,12 +87,10 @@ export class HistoryCompletionSource extends Completions.CompletionSourceFuse {
...
-        this.options.forEach(option => {
-            if (lastFocused !== undefined && lastFocused.value === option.value) {
-                this.select(option)
-            } else
-                option.state = "normal"
-        })
+        this.options.forEach(option => option.state = "normal")
+        for (const option of this.options) {
+            if (lastFocused !== undefined && lastFocused.value === option.value) this.select(option)
+        }
```
Comment (cmcaine) [inline src/completions/History.ts]: You probably meant to put a break in there ;)

As it is, I don't think you actually changed the functionality.

Comment (bovine3dom) [reply src/completions/History.ts]: Wow

**Pull request merged**